src/sentry/api/endpoints/organization_shortids.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from django.db import transaction
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.models import Project
+from sentry.utils.strings import validate_callsign
+
+
+class ShortIdsUpdateEndpoint(OrganizationEndpoint):
+
+    def put(self, request, organization):
+        """
+        Update Short IDs
+        ````````````````
+
+        Updates the call signs of projects within the organization.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          short ID should be looked up in.
+        :param callsigns: a dictionary of project IDs to their intended
+                          callsigns.
+        :auth: required
+        """
+        callsigns = request.DATA.get('callsigns', {})
+        for project_id, callsign in callsigns.iteritems():
+            callsign = validate_callsign(callsign)
+            if callsign is None:
+                return Response({'detail': 'invalid callsign "%s"' % callsign},
+                                status=400)
+            callsigns[project_id] = callsign
+
+        if len(callsigns) != len(callsigns.values()):
```

Comment:
Isn't this testing that the number of keys in the mapping is equal to the number of values…? I'm not sure how this would happen — is this supposed to be `len(set(callsigns.values()))` or am I missing something?